### PR TITLE
View

### PR DESCRIPTION
**Pull request by Masouma96**
Use preprocessed HTML file with embedded Ruby code.
Use layouts and templates for shared content.

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -1,5 +1,9 @@
 class UsersController < ApplicationController
-  def index; end
+  def index
+    @users = User.all
+  end
 
-  def show; end
+  def show
+    @users = User.all
+  end
```
Comment (Deepakdanger) [inline app/controllers/users_controller.rb]: - [ ] Kindly provide here the particular user, not the all user.for ex:
```
def show
  @user = User.find(params[:id])
end
```

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -1,2 +1,44 @@
...
+  <%= link_to user_post_path(@user, id: post:id) do %>
+    <p> Post <% post.id %></p>
+    <p><%= post.text %></p>
+    <p> 
+      <span>
+      Comments: <%= post.comments_counter || 0 %>,
+      Likes: <%= post.likes_counter || 0 %>
+      </span>
+    </p>
+  <% end %>
```
Comment (Deepakdanger) [inline app/views/users/show.html.erb]: - [ ] Here I am getting this error, Kindly resolve this.
Screenshot from 2023-02-04 19-38-14

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @Masouma864 , 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required Changes ♻️
- [ ] While creating the Post there is one error ` undefined method Title for #<Post id: nil, title: "woo", text: "This is my first post", comment_counter: 5, like_counter: 2, created_at: nil, updated_at: nil, user_id: 1>`. There is typo error in post-title validation. Kindly fix it.
- [ ] _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/posts/index.html.erb [hunk 3]
```diff
@@ -1,2 +1,8 @@
-<h1> Posts </h1>
-<p> Lists all posts for Joseph</p>
+<div class="post-container">
+  <%= render partial: 'shared/user-card', locals: { user: @user } %>
+  <% @user.recent_posts.each_with_index do |post, index| %>
+    <%= link_to post_id_path(@user, post), class: 'post-card-link' do %>
+      <%= render 'shared/post-card', post: post, id: post.id, brief: true, comment: true, all: false %>
+    <% end %>
+  <% end %>
```
Comment (omar-muhamad) [inline app/views/posts/index.html.erb]: - [ ] You rendered the `post-card` partial here, As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Views_project.md#:~:text=Use%20methods%20that%20you%20have%20created%20in%20the%20Blog%20App%20%2D%20processing%20data%20in%20models.), Please use the `recent_comments` method you created in the post model to render the most recent 5 comments only here in `posts index` view.

File: app/views/posts/show.html.erb [hunk 4]
```diff
@@ -1,2 +1,3 @@
-<h1>Posts</h1>
-<p> Shows info for Joseph posts </p>
+<div class="post-container">
+  <%= render 'shared/post-card', post: @post, id: @post.id, brief: false, comment: true, all: true %>
+</div>
```
Comment (omar-muhamad) [inline app/views/posts/show.html.erb:3]: - [ ] Great job using partials here.

File: app/views/users/index.html.erb [hunk 5]
```diff
@@ -1,2 +1,11 @@
+<%= stylesheet_link_tag 'index', media: 'all', 'data-turbolinks-track': 'reload' %>
 <h1>Users</h1>
-<p>Lists all users</p>
+<div class="user-list d-flex flex-wrap">
+  <% @all_users.each do |user| %>
+    <div class="user-card m-2">
+      <%= link_to user_id_path(user.id) do %>
+        <%= render partial: 'shared/user-card', locals: { user: user } %>
+      <% end %>
```
Comment (omar-muhamad) [inline app/views/users/index.html.erb:8]: - [ ] Great job using partials here.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @Masouma864,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
Hi @Masouma864, 

### Good job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark:  Clean and organized code
:heavy_check_mark: Descriptive Readme file

Your project is complete! Great work on making the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can get to me on slack  using this handle @mihndim.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**